standards/managing-services-interactions.md

```diff
@@ -0,0 +1,45 @@
+---
+category: Building software
+expires: 2018-03-29
+---
+# Managing interactions between services
+
+Services need to interact with one another over APIs. They need to be
+able to find one another, authenticate clients, be fault-tolerant,
+scale appropriately in line with demand, and more.
+
+## Principles
+
+Services should be built on the assumption that [whatever network they
+reside in is
+insecure](https://en.wikipedia.org/wiki/Fallacies_of_distributed_computing).
+It should be possible to deploy a new instance of a service to a new
+environment (with appropriate configuration) and have it present an
+appropriately-secured interface to the network.
```

Comment:
Agree with this, but it's outright saying that all inter-service traffic on the LAN should be using TLS, which brings significant ops overhead along for the ride - are we ready for that?

Reply:
I think that depends how one reads this? For new services, I'd argue we ought to be using TLS on all inter-service traffic. For old services, this is something we *should* do, but may not?

Reply:
Really I'm just checking that we're aware of the implications of introducing internal TLS really. Off the top of my head I'm not aware of any projects that are doing that currently, are there any?

Reply:
Tax Tribs - the various service components (data capture, file upload, file download, payments) all communicate via https.